lib/services/fanout.go

```diff
@@ -312,6 +312,20 @@ func (f *Fanout) removeWatcher(w *fanoutWatcher) {
 	}
 }
 
+// reorderingDelete efficiently removes an element from a slice via reordering
+// and zeroes the newly out of bounds trailing slot, returning the resized slice.
+func reorderingDelete[T any](s []T, i int) []T {
```

Comment:
I wouldn't be opposed to putting this in `api/utils/slices.go` either.

Reply:
opting against for the time being since this is only needed in v13 at the moment.